src/Dapr.Workflow/WorkflowContext.cs

```diff
@@ -44,47 +44,299 @@ internal WorkflowContext(TaskOrchestrationContext innerContext)
         /// <summary>
         /// Gets the current workflow time in UTC.
         /// </summary>
+        /// <remarks>
+        /// The current workflow time is stored in the workflow history and this API will
+        /// return the same value each time it is called from a particular point in the workflow's
+        /// execution. It is a deterministic, replay-safe replacement for existing .NET APIs for getting
+        /// the current time, such as <see cref="DateTime.UtcNow"/> and <see cref="DateTimeOffset.UtcNow"/>
+        /// (which should not be used).
+        /// </remarks>
         public DateTime CurrentUtcDateTime => this.innerContext.CurrentUtcDateTime;
 
         /// <summary>
-        /// Assigns a custom status value to the current workflow.
+        /// Gets a value indicating whether the workflow is currently replaying a previous execution.
         /// </summary>
-        public void SetCustomStatus(object? customStatus) => this.innerContext.SetCustomStatus(customStatus);
+        /// <remarks>
+        /// <para>
+        /// Workflow functions are "replayed" after being unloaded from memory to reconstruct local variable state.
+        /// During a replay, previously executed tasks will be completed automatically with previously seen values
+        /// that are stored in the workflow history. One the workflow reaches the point in the workflow logic
+        /// where it's no longer replaying existing history, the <see cref="IsReplaying"/> property will return <c>false</c>.
+        /// </para><para>
+        /// You can use this property if you have logic that needs to run only when <em>not</em> replaying. For example,
+        /// certain types of application logging may become too noisy when duplicated as part of replay. The
+        /// application code could check to see whether the function is being replayed and then issue the log statements
+        /// when this value is <c>false</c>.
+        /// </para>
+        /// </remarks>
+        /// <value>
+        /// <c>true</c> if the workflow is currently replaying a previous execution; otherwise <c>false</c>.
+        /// </value>
+        public bool IsReplaying => this.innerContext.IsReplaying;
```

Comment:
I'm not necessarily opposed to this. However, doesn't giving this kind of violate the deterministic nature of the workflows? Or at least the one we're encouraging? Cause this is giving users the ability to have workflows diverge on if it's the first execution or not?

Or am I misunderstanding the purpose of this?

Reply:
You're right that this could be used incorrectly, but I haven't seen that happen in practice with Durable Functions over the past several years. It's mostly used for things like logging, where you only want to write a log statement the first time a piece of code executes.